ui/script.js

```diff
@@ -669,16 +662,23 @@ function helpOff () {
 
 // checksums
 function getSum (type) {
-  sumCall(getASelected().innerText, type)
+  upBarPc.style.display = 'block'
+  upBarPc.innerText = 'computing checksum...'
+  upBarPc.style.width = '100%'
   sumsOff()
+  sumCall(getASelected().innerText, type, loaded => {
```

Comment:
I think that answers your comment wrt the callback - I just get back the target XHR element from the callback and fetch the hash from here.

Also added a message in the upload bar & a flicker when it's computed :+1: